pkg/git/WebhookEventService.go

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2020 Devtron Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package git
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/devtron-labs/git-sensor/internal"
+	"github.com/devtron-labs/git-sensor/internal/sql"
+	"github.com/nats-io/stan"
+	"go.uber.org/zap"
+	_ "gopkg.in/robfig/cron.v3"
+	"regexp"
+)
+
+type WebhookEventService interface {
+	GetAllGitHostWebhookEventByGitHostId(gitHostId int)	([]*sql.GitHostWebhookEvent, error)
+	SaveWebhookEventData(webhookEventData *sql.WebhookEventData) error
+	GetWebhookParsedEventDataByEventIdAndUniqueId(eventId int, uniqueId string) (*sql.WebhookEventParsedData, error)
+	SaveWebhookParsedEventData(webhookEventParsedData *sql.WebhookEventParsedData) error
+	UpdateWebhookParsedEventData(webhookEventParsedData *sql.WebhookEventParsedData) error
+	MatchCiTriggerConditionAndNotify(event *sql.GitHostWebhookEvent, webhookEventParsedData *sql.WebhookEventParsedData, fullDataMap map[string]string) error
+}
+
+type WebhookEventServiceImpl struct {
+	logger                       *zap.SugaredLogger
+	webhookEventRepository       sql.WebhookEventRepository
+	materialRepository           sql.MaterialRepository
+	nats                         stan.Conn
+	gitOperationService			 GitOperationService
+	webhookEventBeanConverter    WebhookEventBeanConverter
+}
+
+func NewWebhookEventServiceImpl (
+	logger *zap.SugaredLogger, webhookEventRepository sql.WebhookEventRepository, materialRepository sql.MaterialRepository, nats stan.Conn,
+	gitOperationService GitOperationService, webhookEventBeanConverter WebhookEventBeanConverter,
```

Comment:
use pointer

Reply:
Taking it as my learning (Will try to corporate from next time onwards. )